lib/config/sources/env_source.rb

```diff
@@ -52,10 +52,28 @@ def load
           leaf[keys.last] = parse_values ? __value(value) : value
         end
 
-        hash
+        convert_hashes_to_arrays(hash)
```

Comment:
These changes look good!

The only change I'd like to see if a new option that enables this behavior (similar to `env_parse_values`) and to set it to `false` by default so it's not likely to break anyone's existing config.

```ruby
Config.setup do |config|
  config.use_env = true
  config.env_parse_arrays = true
end
```